### PR TITLE
bpo-32729 socket error handling change (GH-5458)

### PR DESCRIPTION
**Pull request by rkdls**
Error occurred ```tuple index out of range``` which is obvious bug I think. 

thank you

https://bugs.python.org/issue32729

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

**Comment by WildCard65** [comment]
This belongs on bugs.python.org

**Comment by rkdls** [comment]
I added bpo issue (32729) and signed up CLA.

**Comment by WildCard65** [comment]
@rkdls Remove the "[3.x] " bit from the PR's title and it'll satisfy the issue number for @bedevere-bot

**Comment by Mariatta** [comment]
Thanks. In the PR title, the pattern that bedevere will recognize is `bpo-NNNN` such as `bpo-32729` .
If you edit yours to remove the space around the dash, then it'll satisfy the issue number check

**Comment by rkdls** [comment]
Thanks for answering @Mariatta  @WildCard65 .
I changed. anything i should do ?

**Comment by WildCard65** [comment]
The bottom of this PR lists all the requirements your PR needs.

File: Lib/socket.py [hunk 1]
```diff
@@ -591,8 +591,9 @@ def readinto(self, b):
                 self._timeout_occurred = True
                 raise
             except error as e:
-                if e.args[0] in _blocking_errnos:
-                    return None
+                if e.args:
```
Comment (tiran) [inline Lib/socket.py]: You can write the check in one line, ``if e.args and e.args[0] in _blocking_errnos:``

**Review by tiran — CHANGES_REQUESTED** [review]
See inline comment

You also have to create a ``Misc/News.d`` entry, see https://devguide.python.org/committing/#what-s-new-and-news-entries

File: Lib/socket.py (hunk 1 above)
Comment (rkdls) [reply Lib/socket.py]: you right.
I changed. Thank you

**Comment by rkdls** [comment]
I have made the requested changes; please review again

File: Lib/socket.py [hunk 2]
```diff
@@ -591,7 +591,8 @@ def readinto(self, b):
                 self._timeout_occurred = True
                 raise
             except error as e:
-                if e.args[0] in _blocking_errnos:
+
```
Comment (tiran) [inline Lib/socket.py]: Please drop the surplus newline

**Comment by rkdls** [comment]
@tiran Thanks for commenting. 
deleted line.

File: Lib/socket.py [hunk 3]
```diff
@@ -591,7 +591,7 @@ def readinto(self, b):
                 self._timeout_occurred = True
                 raise
             except error as e:
-                if e.args[0] in _blocking_errnos:
+                if e.args and e.args[0] in _blocking_errnos:
```
Comment (serhiy-storchaka) [inline Lib/socket.py:594]: Why not use `e.errno`?

In what cases `e.args` is empty? Maybe this is an error which should be fixed?

Comment (rkdls) [reply Lib/socket.py:594]: I check out the error Msg, but nothing is get.

But if i check the error type like bellow, 


It is ```TimeoutError``` which is not caught in previous timeout except clause.


The empty case what you asked me is a TimeoutError not caught in timeout previous except clause

Comment (serhiy-storchaka) [reply Lib/socket.py:594]: Please don't use images for text information. They are hard to read by blind people and are disappeared in email clients.

Comment (serhiy-storchaka) [reply Lib/socket.py:594]: Thus the actual bug is that TimeoutError is not caught in previous except clause?

Comment (rkdls) [reply Lib/socket.py:594]: Oh sorry for images.
and yes. you right.  TimeoutError not caught

**Comment by rkdls** [comment]
It's not a handling issue.

**Pull request closed without merging**